src/clib/lib/config/config_parser.cpp

```diff
@@ -435,6 +436,12 @@ bool config_parser_add_key_values(
     return new_node != NULL;
 }
 
+static std::optional<std::string> char_to_optstr(char *p) {
```

Comment:
As a user of this function, i'd expect it not to free my string when converting?

Reply:
Yea, I changed it to take unique_ptr as pass by value so that the expectation is that it takes ownership.